### PR TITLE
Update endpoint dialTimeouts to reasonable values

### PR DESCRIPTION
**Pull request by ArtfulCoder**
This should help clear up unused File Descriptors sooner.

### CODE REVIEW

File: pkg/proxy/userspace/proxysocket.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func newProxySocket(protocol api.Protocol, ip net.IP, port int) (proxySocket, er
 }
 
 // How long we wait for a connection to a backend in seconds
-var endpointDialTimeout = []time.Duration{1, 2, 4, 8}
+var endpointDialTimeout = []time.Duration{250 * time.Millisecond, 500 * time.Millisecond, 2 * time.Second, 4 * time.Second}
```
Comment (thockin) [inline pkg/proxy/userspace/proxysocket.go]: shouldn't this be 250ms, 512ms, 1s, 2s ?  You skipped 1s.

Comment (ArtfulCoder) [inline pkg/proxy/userspace/proxysocket.go]: I wanted to have a value higher than 2s to cover the 2s rtimeout on skydns.
We can do 250ms, 512ms, 1s, 3s ?

Comment (thockin) [inline pkg/proxy/userspace/proxysocket.go]: I'm not sure I understand why that is important?

Comment (ArtfulCoder) [inline pkg/proxy/userspace/proxysocket.go]: ah, I have been dealing with readtimeouts way more than I want to :)
this is dialtimeout :)
will update PR

**Comment by ArtfulCoder** [comment]
Updated PR with your feedback. thanks for the review!

**Comment by thockin** [comment]
Are you going to cherry pick this?

**Comment by ArtfulCoder** [comment]
yes, I want to get this for 1.1
I am hoping it will help the proxier perform better :)

**Comment by thockin** [comment]
I think it's going to be minor, but the change is small, so OK

**Comment by ArtfulCoder** [comment]
we are essentially guarding against endpoints which cant deal with load, get sluggish and cause kube-proxy to try for longer than it should.

File: pkg/proxy/userspace/proxysocket.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func newProxySocket(protocol api.Protocol, ip net.IP, port int) (proxySocket, er
 }
 
 // How long we wait for a connection to a backend in seconds
-var endpointDialTimeout = []time.Duration{1, 2, 4, 8}
+var endpointDialTimeout = []time.Duration{250 * time.Millisecond, 512 * time.Millisecond, 1 * time.Second, 2 * time.Second}
```
Comment (hw-qiaolei) [inline pkg/proxy/userspace/proxysocket.go]: s/512/500. It looks a little strange here.

Comment (wojtek-t) [inline pkg/proxy/userspace/proxysocket.go]: +1

**Comment by ArtfulCoder** [comment]
Please re-review

**Pull request merged**